src/app/parcels/parcelsTable/conditionalStyling.ts

```diff
@@ -0,0 +1,68 @@
+import { DefaultTheme } from "styled-components/dist/types";
+import { ParcelsTableRow } from "./types";
+
+const getRowBoundaries = (columnValues: (string | number | null)[]): number[] => {
+    let temp = columnValues[0];
+    const boundaries: number[] = [];
+
+    columnValues.forEach((columnValue, index) => {
+        if (columnValue !== temp) {
+            boundaries.push(index);
+        }
+        temp = columnValue;
+    });
+
+    return boundaries;
+};
+
+const searchForPackingSlotBreakPoints = (parcelsTableRows: ParcelsTableRow[]): number[] => {
+    const packingSlotValues = parcelsTableRows.map((row) => row.packingSlot);
+    return getRowBoundaries(packingSlotValues);
+};
+
+const searchForPackingDateBreakPoints = (parcelsTableRows: ParcelsTableRow[]): number[] => {
+    const packingSlotValues = parcelsTableRows.map((row) => row.packingDate?.getDate() ?? null);
+    return getRowBoundaries(packingSlotValues);
+};
+
+export type BreakPointConfig = {
```

Comment:
this type is relevant to the table component and would work for any table, so let's put this in the table component rather than the parcelsTable folder